packages/dashboard-core-plugins/src/panels/NotebookPanel.tsx

```diff
@@ -600,6 +601,18 @@ class NotebookPanel extends Component<NotebookPanelProps, NotebookPanelState> {
     this.setState({ showCloseModal: false });
   }
 
+  /**
+   * Closes overwritten tabs
+   * @param fileName The name of the file to be overwritten
+   */
+  handleOverwrite(fileName: string): void {
+    const { glEventHub } = this.props;
+    const { panelState } = this.state;
+
+    glEventHub.emit(NotebookEvent.RENAME_FILE, fileName, fileName, panelState);
```

Comment:
Not sure why this works - we're emitting the same `fileName` for both the old name and new name, which isn't how this event should be used. It should be emitting `oldName`, `newName`, shouldn't need to pass the `panelState` as well...
Thinking about this a bit further - I guess it's not always a rename, since you could be saving a file that hasn't been saved yet (i.e. just writing a new file).
So emitting a `NotebookEvent.CLOSE_FILE` with the `newName` would be more appropriate - really we just want to close any files that are already open with the new name anyway.
Try that out. One edge case we need to be concerned about is if the old file that's already open has been modified, since it may prompt to save/discard changes if we try and close it (which we've agreed we should probably just ignore and close the file).

Reply:
This works with files that have already been saved but I'm not sure how to go about this with unsaved files. For example, if you make a file called "foo.py" and make another file and try to save it as "foo.py", since "foo.py" is not saved, it doesn't have an id so the handler for CLOSE_FILE doesn't have any record of it.

Reply:
I think the behaviour you have is correct :+1: